File: keprix_sdk/typescript/src/domain.ts
export interface Field {
  name: string;
  type?: string;
  required?: boolean;
  default?: unknown;
  entity?: string;
  values?: string[];
}

export interface Operation {
  name: string;
  confirmation_required?: boolean;
}

export interface Entity {
  name: string;
  fields?: Field[];
  operations?: Operation[];
}

export interface Domain {
  name: string;
  entities?: Entity[];
}

File: keprix_sdk/python/src/keprix_sdk/domain.py
from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any


@dataclass
class Field:
    name: str
    type: str = "string"
    required: bool = False
    default: Any = None
    entity: str | None = None
    values: list[str] | None = None


@dataclass
class Operation:
    name: str
    confirmation_required: bool = False


@dataclass
class Entity:
    name: str
    fields: list[Field] = field(default_factory=list)
    operations: list[Operation] = field(default_factory=list)


@dataclass
class Domain:
    name: str
    entities: list[Entity] = field(default_factory=list)
